bin/deploy

```diff
@@ -169,23 +177,52 @@ main() {
   local conda_packages
   conda_packages="conda${pyver_prefix}_${deploy_mode}-${pkg_postfix}.yml"
 
+  if [[ $FIXED_ENVREF != '' ]]; then
+    # (try to) get latest branch SHA1 and assign it to LSST_SPLENV_REF
+    # if the REF is not a branch (or a tag), the reftip will be empty
+    rawreftip=$(git ls-remote https://github.com/lsst/scipipe_conda_env.git ${ENVREF})
+    reftip=$(echo "$rawreftip" | cut -c1-7)
+    reftype=$( echo "$rawreftip" | cut -f 2 -d ' ' | cut -f 2 -d'/' )
+    echo ">> $reftype"
+    if [ "$reftip" == '' ] || [ "$reftype" == 'tags' ]; then
+        # the provided ref is not a branch
+        LSST_SPLENV_REF="${ENVREF}"
+        # Defining environment name based on the provided SHA1
+        LSST_CONDA_ENV_NAME="lsst-scipipe-${FIXED_ENVREF}"
+        local env_file="${LSSTSW}/env/${ENVREF}/${conda_packages}"
+    else
+        # the provided ref is a branch
+        LSST_SPLENV_REF="$reftip"
+        # Defining environment name based on branch and SHA1 from the tip of the branch
+        LSST_CONDA_ENV_NAME="scipipe-env-${FIXED_ENVREF}.${LSST_SPLENV_REF}"
+        local env_file="${LSSTSW}/env/${FIXED_ENVREF}/${reftip}/${conda_packages}"
+    fi
+  else
+    LSST_SPLENV_REF=${LSST_SPLENV_REF:-1172c30}
```

Comment:
Moving the default env SHA1 from line 25 to line 201 makes it much much harder to notice.  I think you can move this line back to line 25 and the logic will still work properly won't it?

Reply:
Moved back to line 23